services/dynamic-sidecar/src/simcore_service_dynamic_sidecar/api/containers_tasks.py

```diff
@@ -0,0 +1,317 @@
+
+
+@containers_router_tasks.post(
+    "/containers",
+    summary=dedent(
+        """
+    Starts a background task responsible for:
+    - cleaning up resources from previous runs
+    - pulling the images
+    - starting the containers
+
+    NOTE: only one instance of this task can run at a time
+    """
+    ).strip(),
+    status_code=status.HTTP_202_ACCEPTED,
+    response_model=TaskId,
+    responses={
+        status.HTTP_422_UNPROCESSABLE_ENTITY: {
+            "description": "Cannot validate submitted compose spec"
+        }
+    },
+)
+@cancel_on_disconnect
+async def create_service_containers_task(  # pylint: disable=too-many-arguments
+    request: Request,
+    containers_create: ContainersCreate,
+    task_manager: TaskManager = Depends(get_task_manager),
+    settings: ApplicationSettings = Depends(get_settings),
+    shared_store: SharedStore = Depends(get_shared_store),
+    app: FastAPI = Depends(get_application),
+    application_health: ApplicationHealth = Depends(get_application_health),
+    rabbitmq: RabbitMQ = Depends(get_rabbitmq),
+    mounted_volumes: MountedVolumes = Depends(get_mounted_volumes),
+) -> TaskId:
+    assert request  # nosec
+
+    try:
+        task_id = start_task(
+            task_manager=task_manager,
+            handler=task_create_service_containers,
+            unique=True,
+            settings=settings,
+            containers_create=containers_create,
+            shared_store=shared_store,
+            mounted_volumes=mounted_volumes,
+            app=app,
+            application_health=application_health,
+            rabbitmq=rabbitmq,
+        )
+        return task_id
+    except TaskAlreadyRunningError as e:
+        raise HTTPException(status.HTTP_409_CONFLICT, detail=f"{e}") from e
+
+
+@containers_router_tasks.post(
+    "/containers:down",
+    summary="Removes the previously started service and returns the docker-compose output",
+    status_code=status.HTTP_202_ACCEPTED,
+    response_model=TaskId,
+)
+@cancel_on_disconnect
+async def runs_docker_compose_down_task(
+    request: Request,
+    task_manager: TaskManager = Depends(get_task_manager),
+    settings: ApplicationSettings = Depends(get_settings),
+    shared_store: SharedStore = Depends(get_shared_store),
+    app: FastAPI = Depends(get_application),
+) -> TaskId:
+    assert request  # nosec
+
+    try:
+        task_id = start_task(
+            task_manager=task_manager,
+            handler=task_runs_docker_compose_down,
+            unique=True,
+            app=app,
+            shared_store=shared_store,
+            settings=settings,
+        )
+        return task_id
+    except TaskAlreadyRunningError as e:
+        raise HTTPException(status.HTTP_409_CONFLICT, detail=f"{e}") from e
+
+
+@containers_router_tasks.post(
+    "/containers/state:restore",
+    summary=dedent(
+        """
+    Restores the state of the dynamic service
+
+    When restoring the state:
+    - pull inputs via nodeports
```

Comment:
suggestion: - pull inputs

Reply:
and one question, why does this entrypoint do 2 things? (pulling inputs and state) vs the following entrypoints doing only one thing? or is this an error?

Reply:
Good catch it only restores the state. docstring was wrong